### PR TITLE
chore(compliance): add "missing" and "n/a" test statuses

### PR DESCRIPTION
**Pull request by eladb**
Compliance tests now can have one of the following 4 statuses:

- 🟢 - Test passes for this language.
- ⚪ - Test is not applicable for this language.
- ⭕ - Test is not implemented (yet) for this language
- 🔴 - Test is failing for this language: missing feature or bug in language bindings.

To facilitate reporting these statuses, the Go compliance suite now now has two methods:

 - `suite.FailTest(reason, url)` - reports a test as a failing and requires a URL for a github issue that will be rendered as a link in the markdown result.
 - `suite.NotApplicableTest(reason)` - reports a test as N/A for this specific language bindings.

The test result can now include two additional optional fields: `reason` and `url`. Currently, only `url` is taken into account and rendered in the markdown.

The "exclusions" mechanism was completely removed from the compliance report model.

---

By submitting this pull request, I confirm that my contribution is made under the terms of the [Apache 2.0 license].

[Apache 2.0 license]: https://www.apache.org/licenses/LICENSE-2.0

### CODE REVIEW

File: tools/jsii-compliance/report.ts [hunk 1]
```diff
@@ -9,27 +9,10 @@ import { suite } from './suite';
...
-  suite: schema.Suite,
-  language: string,
-) {
-  const testExcluded = !!testCase.exclusions?.[language];
-  const bindingExcluded = !!suite.exclusions?.[language];
-  return testExcluded || bindingExcluded;
-}
+const SUCCESS = '✅'; // test succeeded
+const FAILURE = '❌'; // test is failing
+const MISSING = ''; // test is not implemented yet
```
Comment (iliapolo) [inline tools/jsii-compliance/report.ts:14]: Would add some text here, e.g `missing`, or a question mark emoji or something.

And should also be reflected at the bottom of https://github.com/aws/jsii/tree/main/tools/jsii-compliance which explains the report statuses.

File: tools/jsii-compliance/suite.ts [hunk 2]
```diff
@@ -18,612 +18,480 @@ export const suite: schema.Suite = {
...
-      exclusions: {},
     },
     {
       name: 'createObjectAndCtorOverloads',
       description:
         'Class can be instantiated with empty and non empty constructor',
-      exclusions: {
-        golang: {
-          reason:
-            'Golang does not have overloaded functions so the genearated class only has a single New function.',
```
Comment (iliapolo) [inline tools/jsii-compliance/suite.ts:67]: Should be migrated to the new mechanism.

**Review by iliapolo — APPROVED** [review]
Cool. Some minor stuff. Added `do-not-merge`.

File: tools/jsii-compliance/report.ts (hunk 1 above)
Comment (eladb) [reply tools/jsii-compliance/report.ts:14]: Changed to:

- 🟢 - Test passes for this language.
- ⚪ - Test is not applicable for this language.
- ⭕ - Test is not implemented (yet) for this language
- 🔴 - Test is failing for this language: missing feature or bug in language bindings.

File: tools/jsii-compliance/suite.ts (hunk 2 above)
Comment (eladb) [reply tools/jsii-compliance/suite.ts:67]: DONE

**Pull request merged**